lib/ansible/module_utils/common/text/converters.py

```diff
@@ -23,7 +26,27 @@
                                       'surrogate_then_replace'))
 
 
-def to_bytes(obj, encoding='utf-8', errors=None, nonstring='simplerepr'):
+@_t.overload
+def to_bytes(
+    obj: _T,
+    encoding: str = 'utf-8',
+    errors: str | None = None,
+    *,
+    nonstring: _t.Literal['passthru'],
+) -> _T | bytes: ...
+
+
+@_t.overload
+def to_bytes(
+    obj: _t.Any,
```

Comment:
Any reason not to use
```suggestion
    obj: object,
```
?

Reply:
Yeah, for this case I deemed it potentially problematic to use `object` because we cannot necessarily gaurantee that "objects" sent actually inherit from `object`, such as certain C extension types. I know that the python-apt C extension has problematic types, so figured this was a better choice here.

Reply:
Wouldn't those still inherit from `PyObject`, though? I don't know that it's possible to have something in the Python land that doesn't inherit from `object`. Do you have an example?
We don't really need to guarantee such things anyway as it'd be the job of the type checker to verify. But again, I don't see how the situation you describe is at all possible.